server/service/apple_mdm.go

```diff
@@ -491,7 +487,7 @@ func (svc *Service) NewMDMAppleConfigProfile(ctx context.Context, teamID uint, r
 	return newCP, nil
 }
 
-func validateConfigProfileFleetVariables(appConfig *fleet.AppConfig, contents string, lic *fleet.LicenseInfo) (map[string]struct{}, error) {
+func validateConfigProfileFleetVariables(ctx context.Context, ds fleet.Datastore, contents string, lic *fleet.LicenseInfo) (map[string]struct{}, error) {
```

Comment:
I think it might be cleaner if we have helper functions like `validateConfigProfileFleetVariables` and `preprocessProfileContents` take `GroupedCertificateAuthorities` as a param and make the caller of the helper function responsible for getting the grouped CAs from the datastore. And for convenience, the datastore can include a separate method for `GetGroupedCertificateAuthorities` that both gets and transforms the grouped CAs (i.e. lines 501-508).